pkg/controller/datavolume/snapshot-clone-controller.go

```diff
@@ -180,83 +193,88 @@ func (r *SnapshotCloneReconciler) syncSnapshotClone(log logr.Logger, req reconci
 		return syncRes, nil
 	}
 
-	// Check if source snapshot exists and do proper validation before attempting to clone
-	if done, err := r.validateCloneAndSourceSnapshot(&syncRes); err != nil {
-		return syncRes, err
-	} else if !done {
-		return syncRes, nil
-	}
-
-	nn := types.NamespacedName{Namespace: datavolume.Spec.Source.Snapshot.Namespace, Name: datavolume.Spec.Source.Snapshot.Name}
-	snapshot := &snapshotv1.VolumeSnapshot{}
-	if err := r.client.Get(context.TODO(), nn, snapshot); err != nil {
-		return syncRes, err
-	}
-
-	valid, err := r.isSnapshotValidForClone(snapshot)
-	if err != nil || !valid {
-		return syncRes, err
-	}
-
-	fallBackToHostAssisted, err := r.evaluateFallBackToHostAssistedNeeded(datavolume, pvcSpec, snapshot)
-	if err != nil {
-		return syncRes, err
-	}
-
 	if pvc == nil {
-		if !fallBackToHostAssisted {
-			res, err := r.reconcileRestoreSnapshot(log, datavolume, snapshot, pvcSpec, transferName, &syncRes)
-			syncRes.result = &res
-			return syncRes, err
+		pvcModifier := r.updateAnnotations
+		if syncRes.usePopulator {
+			if isCrossNamespaceClone(datavolume) {
+				if !cc.HasFinalizer(datavolume, crossNamespaceFinalizer) {
+					cc.AddFinalizer(datavolume, crossNamespaceFinalizer)
+					return syncRes, r.syncCloneStatusPhase(&syncRes, cdiv1.CloneScheduled, nil)
+				}
+			}
+			pvcModifier = r.updatePVCForPopulation
+		} else {
+			if done, err := r.validateAndInitLegacyClone(&syncRes); err != nil {
+				return syncRes, err
+			} else if !done {
+				return syncRes, nil
+			}
 		}
 
-		if err := r.createTempHostAssistedSourcePvc(datavolume, snapshot, pvcSpec, &syncRes); err != nil {
-			return syncRes, err
-		}
-		targetHostAssistedPvc, err := r.createPvcForDatavolume(datavolume, pvcSpec, r.updateAnnotations)
+		targetPvc, err := r.createPvcForDatavolume(datavolume, pvcSpec, pvcModifier)
```

Comment:
I went ahead and debugged the sizeless clone failure from volumesnapshot clones:
Since we now create the target PVC through createPvcForDatavolume, nothing takes
care of filling out the missing size
```
spec.resources[storage]: Invalid value: \"0\": must be greater than zero
```
I think we can simply have something similar to this func https://github.com/kubevirt/containerized-data-importer/blob/75e1297ed81fb9aef7d8b951992a7b9c4b58f2f1/pkg/controller/datavolume/pvc-clone-controller.go#L501
(but way simpler, just grabs the size over from the snapshot)